docs/source/design/user_messaging.rst

```diff
@@ -0,0 +1,124 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_user_messaging:
+
+*******************************************************
+User messaging: result records vs exceptions vs logging
+*******************************************************
+
+.. topic:: Specification scope and status
+
+   This specification aims to delineate the applicable contexts for using result records,
+   exceptions, logging, or other types of user messaging methods.
```

Comment:
what @mih above says is to use language similar to the already present entries, where now I think
```suggestion
   This specification provides a partial overview of the current implementation.
```